dev/tests/unit/Magento/FunctionalTestFramework/Util/ModuleResolverTest.php

```diff
@@ -90,141 +103,130 @@ public function testGetModulePathsAggregate()
+        $property = new ReflectionProperty(ModuleResolver::class, 'instance');
         $property->setAccessible(true);
         $property->setValue(null);
 
         $this->mockForceGenerate(false);
-        $mockResolver = $this->setMockResolverClass(
-            true,
-            [],
-            null,
-            null,
-            [],
-            [],
-            [],
-            null,
-            null,
-            [],
-            [],
-            null,
-            function ($arg) {
-                return $arg;
-            },
-            function ($arg) {
-                return $arg;
-            }
-        );
-        $resolver = ModuleResolver::getInstance();
-        $this->setMockResolverProperties($resolver, null, null);
-        $this->assertEquals(
-            [],
-            $resolver->getModulesPath()
-        );
+        // Define the Module paths from default TESTS_MODULE_PATH
+        $modulePath = defined('TESTS_MODULE_PATH') ? TESTS_MODULE_PATH : TESTS_BP;
 
         // Define the Module paths from app/code
         $magentoBaseCodePath = MAGENTO_BP;
 
-        // Define the Module paths from default TESTS_MODULE_PATH
-        $modulePath = defined('TESTS_MODULE_PATH') ? TESTS_MODULE_PATH : TESTS_BP;
-
-        $mockResolver->verifyInvoked('globRelevantPaths', [$modulePath, '']);
-        $mockResolver->verifyInvoked(
-            'globRelevantPaths',
-            [$magentoBaseCodePath . DIRECTORY_SEPARATOR . "vendor" , 'Test' . DIRECTORY_SEPARATOR .'Mftf']
-        );
-        $mockResolver->verifyInvoked(
-            'globRelevantPaths',
-            [
-                $magentoBaseCodePath . DIRECTORY_SEPARATOR . "app" . DIRECTORY_SEPARATOR . "code",
-                'Test' . DIRECTORY_SEPARATOR .'Mftf'
-            ]
-        );
+        $moduleResolverService = $this->createPartialMock(ModuleResolverService::class, ['globRelevantPaths']);
+        $moduleResolverService
+            ->method('globRelevantPaths')
+            ->withConsecutive()
+            ->will(
+                $this->returnCallback(
+                    function ($codePath, $pattern) use ($modulePath, $magentoBaseCodePath) {
+                        if ($codePath === $modulePath && $pattern === '') {
+                            return [];
+                        }
+
+                        if ($codePath === $magentoBaseCodePath . '/vendor' && $pattern === 'Test/Mftf') {
+                            return [];
+                        }
+
+                        if ($codePath === $magentoBaseCodePath . "/app/code" && $pattern === 'Test/Mftf') {
+                            return [];
+                        }
+
+                        $this->fail(sprintf(
+                            'Not expected parameter: \'%s\' when invoked method globRelevantPaths().',
+                            $modulePath
+                        ));
+                    }
+                )
+            );
```

Comment:
@anzin 
This is close but does not match exactly what the original test does. The original test makes sure that all the paths parameters are used at least once, but the new test will miss that.